tests/integration/test_schema.py

```diff
@@ -863,12 +863,12 @@ async def assert_schema_versions(client: Client, trail: str, schema_id: int, exp
     """
     res = await client.get(f"/schemas/ids/{schema_id}/versions{trail}")
     assert res.status_code == 200
+    registered_schemas = res.json()
 
     # Schema Registry doesn't return an ordered list, Karapace does.
     # Need to check equality ignoring ordering.
-    assert len(res.json()) == len(expected)
-    for e in ({"subject": e[0], "version": e[1]} for e in expected):
-        assert e in res.json()
+    result = [(schema["subject"], schema["version"]) for schema in registered_schemas]
+    assert set(result) == set(expected)
```

Comment:
this is not a fix. Just a better debugging assert (the previous would fail in the length check without giving any useful info on what was different)